src/hooks/useMessageList.js

```diff
@@ -0,0 +1,219 @@
+ * @function updateMessageReadStatus
+ * @param {object} messageObject - An object containing inputs for the the message
+ * @param {Function} fetchData - Inrupt's session.fetch function
+ * @returns {Promise} Promise - Perform action that updates read status of message
+ * on messageObject
+ */
+const updateMessageReadStatus = async (messageObject, fetchData) => {
+  let messageDataset = await getSolidDataset(messageObject.messageUrl, { fetch: fetchData });
+  let messageStatusThing = getThing(messageDataset, `${messageObject.messageUrl}#messagestatus`);
+
+  if (messageStatusThing) {
+    const readStatus = getBoolean(messageStatusThing, RDF_PREDICATES.value);
+
+    if (!readStatus) {
+      messageStatusThing = buildThing(messageStatusThing)
+        .setBoolean(RDF_PREDICATES.value, true)
+        .build();
+    }
+  } else {
+    messageStatusThing = buildThing(createThing({ name: 'messagestatus' }))
+      .addStringNoLocale(RDF_PREDICATES.propertyValue, 'Read Status')
+      .addBoolean(RDF_PREDICATES.value, true)
+      .build();
+  }
+
+  messageDataset = setThing(messageDataset, messageStatusThing);
+
+  try {
+    await saveSolidDatasetAt(messageObject.messageUrl, messageDataset, { fetch: fetchData });
+  } catch (error) {
+    throw new Error('Failed to update ttl file.');
+  }
+};
+
+export { updateMessageReadStatus };
+
+/**
+ * @typedef {object} messageListHookObject
+ * @property {boolean} isLoading - if the message list is loading
+ * @property {boolean} isSuccess - if the message list has been loaded
+ * @property {boolean} isError - if an error occurred while fetching the message
+ * list
+ * @property {object} error - the error that occurred while fetching
+ * @property {Array} data - the list of messageObjects containing the message
+ * content and information
+ * @property {Function} refetch - function to refetch data
+ * @property {boolean} isFetching - is the collection being fetched
+ * @property {(messageObject) => Promise} add - updates the messageObject from messageListObject
+ */
+
+/**
+ * Custom hook that helps renders messages from Inbox/Outbox from PASS container
+ * in Solid. It's a wrapper for a React Query and manages the messageListObject
+ * pulled and parsed from Solid
+ *
+ * @memberof hooks
+ * @function useMessageList
+ * @param {string} messageType - The string for 'Inbox' or 'Outbox'
+ * @returns {messageListHookObject} messageListObject - all the data provided by
+ * the useQuery call
+ */
+const useMessageList = (messageType) => {
+  const { session, podUrl } = useSession();
+  const { fetch } = session;
+  const [storedDataset, setStoredDataset] = useState(null);
+  const queryClient = useQueryClient();
+  const containerUrl = podUrl && new URL(`PASS/${messageType}/`, podUrl).toString();
+  let messageList = [];
+
+  const parse = async (data) => {
+    const solidDataset = getThingAll(data);
+    const allMessagesDataset = solidDataset.filter((thing) => thing.url.endsWith('ttl'));
```

Comment:
Ah so the messages are all stored as separate `.ttl` documents within a pod. We should be able to make the `useRdfCollection` hook support this case. I'll make a note and follow up.